addons/website_livechat/models/website.py

```diff
@@ -44,6 +45,10 @@ def _get_livechat_request_session(self):
                 ('has_message', '=', True)
             ], order='create_date desc', limit=1)
             if chat_request_channel:
+                guest = self.env['mail.guest']._get_guest_from_context()
+                if not guest:
+                    guest = chat_request_channel.channel_member_ids.filtered(lambda m: m.guest_id).guest_id
+                    add_guest_cookie(guest)
```

Comment:
Both partners and guests are added when the channel is created. Here we're in a situation where the current user is the guest member but the cookie was not added (since the channel/guest were created by the operator) so we just add it.